src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -209,6 +229,84 @@ static int ask_password_plymouth(
         return 0;
 }
 
+static void free_consoles(struct console *con, unsigned int num) {
+        unsigned int n;
+
+        if (!con || !num)
+                return;
+
+        for (n = 0; n < num; n++)
+                free(con[n].tty);
+
+        free(con);
+}
+
+static const char *current_dev = "/dev/console";
+
+static int collect_consoles(struct console **consoles, unsigned int *num) {
+        _cleanup_free_ char *active = NULL;
+        const char *word, *state;
+        struct console *con = NULL;
+        size_t con_len = 0, len;
+        unsigned int count = 0;
+        int ret;
+
+        assert(num);
+        assert(consoles);
+
+        ret = read_one_line_file("/sys/class/tty/console/active", &active);
+        if (ret < 0)
+                return ret;
+
+        FOREACH_WORD(word, len, active, state) {
+                _cleanup_free_ char *tty = NULL;
+
+                if (strneq(word, "tty0", len) &&
+                    read_one_line_file("/sys/class/tty/tty0/active", &tty) >= 0) {
+                        word = tty;
+                        len = strlen(tty);
+                }
+
+                con = greedy_realloc((void**)&con, &con_len, 1+count, sizeof(struct console));
```

Comment:
Also greedy_realloc() seems to use realloc(3) which should set  ENOMEM.  Or does systemd use a private realloc() instead of the glibc version?  Beside this, I see mixed usage of greedy_realloc() / GREEDY_REALLOC() error handling in the source tree ... e.g. many times a -ENOMEM is returned but also often a return log_oom() is used.  What is the correct way here?